src/converter/add-class-property-declarations/add-class-property-declarations.ts

```diff
@@ -48,32 +48,109 @@ export function addClassPropertyDeclarations( tsAstProject: Project ): Project {
-			.reduce( ( props: Set<string>, prop: ClassInstancePropertyTypes ) => {
+			.reduce((props: Set<string>, prop: ClassInstancePropertyTypes) => {
 				const propName = prop.getName();
-				return propName ? props.add( propName ) : props;
-			}, new Set<string>() );
+				return propName ? props.add(propName) : props;
+			}, new Set<string>());
 
-		let undeclaredProperties = [ ...jsClassProperties ]
-			.filter( ( propName: string ) => !currentPropertyDeclarations.has( propName ) );
+		let undeclaredProperties = [...jsClassProperties]
+			.filter((propName: string) => !currentPropertyDeclarations.has(propName));
 
 		// If the utility found a reference to this.constructor, we don't want to
 		// add a property called 'constructor'. Filter that out now.
 		// https://github.com/gregjacobs/js-to-ts-converter/issues/9
 		undeclaredProperties = undeclaredProperties
-			.filter( ( propName: string ) => propName !== 'constructor' );
+			.filter((propName: string) => propName !== 'constructor');
 
 		// Add all currently-undeclared properties
-		const propertyDeclarations = undeclaredProperties.map( propertyName => {
+		const propertyDeclarations = undeclaredProperties.map(propertyName => {
+			let [tsType, tsDefault, tsIsNullable, oaType, oaFormat, commentText] = getTypesFromComment(propertyName, classDeclaration);
+
+			if (tsIsNullable === 'true') {
+				tsType = `${tsType} | null`;
+			}
 			return {
 				name: propertyName,
-				type: 'any',
+				type: tsType,
 				scope: Scope.Public
 			} as PropertyDeclarationStructure;
-		} );
+		});
+
+		logger.verbose(`    Adding property declarations for properties: '${undeclaredProperties.join("', '")}'`);
+		classDeclaration.insertProperties(0, propertyDeclarations);
+
+		// logger.verbose(`    Setting property defaults for properties: '${undeclaredProperties.join("', '")}'`);
+		undeclaredProperties.map(propertyName => {
+			let [tsType, tsDefault, tsIsNullable, oaType, oaFormat, commentText] = getTypesFromComment(propertyName, classDeclaration);
 
-		logger.verbose( `    Adding property declarations for properties: '${undeclaredProperties.join( "', '" )}'` );
-		classDeclaration.insertProperties( 0, propertyDeclarations );
-	} );
+			// Add default value to a property
+			const propDeclaration = classDeclaration.getPropertyOrThrow(propertyName)
+			if (propDeclaration !== null) {
+				if (tsDefault !== '') {
+					propDeclaration.setInitializer(tsDefault);
+				}
+				if (commentText !== '') {
+					const comments = propDeclaration.getLeadingCommentRanges();
+					const i = 0;
+					//const initializer = propDeclaration.getInitializerIfKindOrThrow(SyntaxKind.SingleLineCommentTrivia);
+					//initializer.insertProperty(1, commentText);
+				}
+			}
+		});
+	});
 
 	return tsAstProject;
+}
+
+function getTypesFromComment(propertyName: string, classDeclaration: ClassDeclaration): [string, string, string, string, string, string] {
```

Comment:
Somewhat minor, but the return type of `[string, string, string, string, string, string]` as a tuple is a bit difficult to work with, and a developer destructuring it could accidentally mix up the positions of the returned values. Would be better to return an object with named properties such as:

```ts
function getTypesFromComment(propertyName: string, classDeclaration: ClassDeclaration): {
    tsType: string;
    tsDefault: string;
    tsIsNullable: string;
    oaType: string;
    oaFormat: string;
    commentText: string;
} {
    // ... function code here

    return {tsType, tsDefault, tsIsNullable, oaType, oaFormat, commentText};
}
```

Reply:
done